### PR TITLE
SDI-424 Endpoint to delete specific incentive mapping

### PR DESCRIPTION
**Pull request by andymarke**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncentiveMappingResource.kt [hunk 1]
```diff
@@ -67,7 +67,12 @@ class IncentiveMappingResource(private val mappingService: IncentiveMappingServi
       ApiResponse(
         responseCode = "200",
         description = "Mapping Information Returned",
-        content = [Content(mediaType = "application/json", schema = Schema(implementation = IncentiveMappingDto::class))]
+        content = [
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncentiveMappingResource.kt:70]: formatting

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncentiveMappingResource.kt [hunk 2]
```diff
@@ -100,7 +105,12 @@ class IncentiveMappingResource(private val mappingService: IncentiveMappingServi
...
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncentiveMappingResource.kt:108]: formatting

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncentiveMappingResource.kt [hunk 3]
```diff
@@ -143,14 +158,15 @@ class IncentiveMappingResource(private val mappingService: IncentiveMappingServi
...
+  suspend fun getLatestMigratedIncentiveMapping(): IncentiveMappingDto =
+    mappingService.getIncentiveMappingForLatestMigrated()
 
   @PreAuthorize("hasRole('ROLE_NOMIS_INCENTIVES')")
   @DeleteMapping("/mapping/incentives")
   @ResponseStatus(HttpStatus.NO_CONTENT)
   @Operation(
     summary = "Deletes incentive mappings",
-    description = "Deletes all rows from the the incentive mapping table. Requires role ADMIN_MAPPING",
+    description = "Deletes all rows from the the incentive mapping table. Requires role NOMIS_INCENTIVES",
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncentiveMappingResource.kt:169]: correction

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncentiveMappingResourceIntTest.kt [hunk 4]
```diff
@@ -543,7 +543,7 @@ class IncentiveMappingResourceIntTest : IntegrationTestBase() {
 
   @DisplayName("DELETE /mapping/incentives")
   @Nested
-  inner class DeleteMappingTest {
+  inner class DeleteMappingsTest {
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncentiveMappingResourceIntTest.kt:546]: rename, this deletes all mappings

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncentiveMappingResourceIntTest.kt [hunk 5]
```diff
@@ -592,47 +592,136 @@ class IncentiveMappingResourceIntTest : IntegrationTestBase() {
...
 
-  @Test
-  fun `delete incentive mappings - migrated mappings only`() {
-    webTestClient.post().uri("/mapping/incentives")
-      .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_INCENTIVES")))
-      .contentType(MediaType.APPLICATION_JSON)
-      .body(BodyInserters.fromValue(createIncentiveMapping()))
-      .exchange()
-      .expectStatus().isCreated
+    @Test
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncentiveMappingResourceIntTest.kt:596]: moved inside inner class

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**